pkg/tsdb/cloudwatch/time_series_query_test.go

```diff
@@ -136,3 +137,268 @@ func TestTimeSeriesQuery(t *testing.T) {
+						"matchExact": true,
+						"refId": "A"
+					}`),
+			},
+		},
+	})
+
+	assert.NoError(t, err)
+	// asserts that period '300' and stat 'Average' are parsed from the JSON query expression and used in the alias, not the JSON model's 'statistic' nor 'period' fields
+	assert.Equal(t, "300 Average", resp.Responses["A"].Frames[0].Name)
+}
+
+func Test_QueryData_executeTimeSeriesQuery_no_alias_provided_frame_name_is_queryId_when_query_isMathExpression(t *testing.T) {
+	origNewCWClient := NewCWClient
+	t.Cleanup(func() {
+		NewCWClient = origNewCWClient
+	})
+	var cwClient fakeCWClient
+	NewCWClient = func(sess *session.Session) cloudwatchiface.CloudWatchAPI {
+		return &cwClient
+	}
+	cwClient = fakeCWClient{
+		GetMetricDataOutput: cloudwatch.GetMetricDataOutput{
+			MetricDataResults: []*cloudwatch.MetricDataResult{
+				{StatusCode: aws.String("Complete"), Id: aws.String("query id"), Label: aws.String("NetworkOut"),
+					Values: []*float64{aws.Float64(1.0)}, Timestamps: []*time.Time{{}}},
+			},
+		},
+	}
+	im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		return datasourceInfo{}, nil
+	})
+	executor := newExecutor(im, newTestConfig(), &fakeSessionCache{})
+
+	resp, err := executor.QueryData(context.Background(), &backend.QueryDataRequest{
+		PluginContext: backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}},
+		Queries: []backend.DataQuery{
+			{
+				RefID:     "A",
+				TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+				JSON: json.RawMessage(`{
+						"type":      "timeSeriesQuery",
+						"metricQueryType": 0,
+						"metricEditorMode": 1,
+						"namespace": "",
+						"metricName": "",
+						"region": "us-east-2",
+						"id": "query id",
+						"statistic": "Maximum",
+						"period": "1200",
+						"hide": false,
+						"matchExact": true,
+						"refId": "A"
+					}`),
+			},
+		},
+	})
+
+	assert.NoError(t, err)
+	assert.Equal(t, "query id", resp.Responses["A"].Frames[0].Name)
+}
+
+func Test_QueryData_executeTimeSeriesQuery_no_alias_provided_frame_name_depends_on_dimension_values_and_matchExact(t *testing.T) {
+	origNewCWClient := NewCWClient
+	t.Cleanup(func() {
+		NewCWClient = origNewCWClient
+	})
+	var cwClient fakeCWClient
+	NewCWClient = func(sess *session.Session) cloudwatchiface.CloudWatchAPI {
+		return &cwClient
+	}
+	cwClient = fakeCWClient{
+		GetMetricDataOutput: cloudwatch.GetMetricDataOutput{
+			MetricDataResults: []*cloudwatch.MetricDataResult{
+				{StatusCode: aws.String("Complete"), Id: aws.String("query id"), Label: aws.String("response label"),
+					Values: []*float64{aws.Float64(1.0)}, Timestamps: []*time.Time{{}}},
+			},
+		},
+	}
+	im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		return datasourceInfo{}, nil
+	})
+	executor := newExecutor(im, newTestConfig(), &fakeSessionCache{})
```

Comment:
I don't really know how to do this in go, but in js I would do something like:
```
const setup = (inputs specific to each test test) => {
return [things, that, we, need, to, run ,the, actual, test]
}
```
does that feel applicable to you here?

Reply:
Tried to do this now for the query, hopefully it is easier to read